packages/compass-global-writes/src/components/shard-zones-table.tsx

```diff
@@ -8,47 +8,189 @@ import {
   Row,
   Cell,
   css,
+  useLeafyGreenTable,
+  type LGColumnDef,
+  type LeafyGreenTableRow,
+  flexRender,
+  type HeaderGroup,
+  SearchInput,
+  type LGTableDataType,
+  getFilteredRowModel,
+  type LgTableRowType,
 } from '@mongodb-js/compass-components';
 import type { ShardZoneData } from '../store/reducer';
 
 const containerStyles = css({
-  maxWidth: '700px',
   height: '400px',
 });
 
+interface ShardZoneRow {
+  locationName: string;
+  zone: string;
+}
+
+interface ShardZoneExpandableRow extends ShardZoneRow {
+  subRows: ShardZoneRow[];
+}
+
+const columns: Array<LGColumnDef<ShardZoneRow>> = [
+  {
+    accessorKey: 'locationName',
+    header: 'Location Name',
+    enableSorting: true,
+  },
+  {
+    accessorKey: 'zone',
+    header: 'Zone',
+    enableSorting: true,
+    enableGlobalFilter: false,
+  },
+];
+
+const parseRow = ({
+  isoCode,
+  readableName,
+  zoneName,
+  zoneLocations,
+}: ShardZoneData): ShardZoneRow => ({
+  locationName: `${readableName} (${isoCode})`,
+  zone: `${zoneName} (${zoneLocations.join(', ')})`,
+});
+
+const parseData = (shardZones: ShardZoneData[]): ShardZoneExpandableRow[] => {
+  const grouppedZones = shardZones.reduce<
+    Record<ShardZoneData['typeOneIsoCode'], ShardZoneExpandableRow>
+  >((groups, next) => {
+    const { typeOneIsoCode, isoCode } = next;
+    if (!groups[typeOneIsoCode]) {
+      groups[typeOneIsoCode] = { ...parseRow(next), subRows: [] };
+    }
+    if (typeOneIsoCode === isoCode) {
+      Object.assign(groups[typeOneIsoCode], parseRow(next));
```

Comment:
Not sure I understand which case this is handling: wouldn't this already be covered by the initial `groups[typeOneIsoCode]` assignment above?

Reply:
Ah, this detail escaped me when reading through the logic, I see it now, didn't really connect that the parseRow we're doing there might not be the actual "type one" code, makes sense! Do you think the parseRow above is needed then? Are we adding it only to make typescript happy?

Reply:
Almost want to suggest to explicitly assign something like `{ locationName: typeOneIsoCode, zone: typeOneIsoCode }` there just so it's clear that this is sort of a placeholder for the top level zone before we can assign something meaningful there, but I'll leave to you to decide, it's a small thing